core/templates/pages/exploration-editor-page/translation-tab/services/translation-language.service.ts

```diff
@@ -17,38 +17,52 @@
  * in the translation tab is currently active.
  */
 
-import { EventEmitter } from '@angular/core';
-
-require('domain/utilities/language-util.service.ts');
-
-angular.module('oppia').factory('TranslationLanguageService', [
-  '$log', 'LanguageUtilService',
-  function($log, LanguageUtilService) {
-    var activeLanguageCode = null;
-    var allAudioLanguageCodes = (
-      LanguageUtilService.getAllVoiceoverLanguageCodes());
-    var _activeLanguageChangedEventEmitter = new EventEmitter();
-    return {
-      getActiveLanguageCode: function() {
-        return activeLanguageCode;
-      },
-      setActiveLanguageCode: function(newActiveLanguageCode) {
-        if (allAudioLanguageCodes.indexOf(newActiveLanguageCode) < 0) {
-          $log.error('Invalid active language code: ' + newActiveLanguageCode);
-          return;
-        }
-        activeLanguageCode = newActiveLanguageCode;
-        _activeLanguageChangedEventEmitter.emit();
-      },
-      getActiveLanguageDescription: function() {
-        if (!activeLanguageCode) {
-          return null;
-        }
-        return LanguageUtilService.getAudioLanguageDescription(
-          activeLanguageCode);
-      },
-      get onActiveLanguageChanged() {
-        return _activeLanguageChangedEventEmitter;
-      }
-    };
-  }]);
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { EventEmitter, Injectable } from '@angular/core';
+import { LanguageUtilService } from 'domain/utilities/language-util.service';
+import { LoggerService } from 'services/contextual/logger.service';
+import { List } from 'lodash';
```

Comment:
The import order we follow is this:

1. Angular imports (imports from @angular/*)
2. Third-party imports (imports from rxjs, ng bootstrap, etc)
3. Application imports. (imports from our application)

Within these sub-categories: sort lexicographically based on module (i.e. where it is imported from, not what is being imported).
For example:
#### Right way (because 'B' < 'D')
```
import { C } from 'B';
import { A } from 'D'; 
```

#### Wrong way
```
import { A } from 'D';
import { C } from 'B'; 
```